internal/dag/serviceapis_processor.go

```diff
@@ -26,5 +35,126 @@ func (p *ServiceAPIsProcessor) Run(dag *DAG, source *KubernetesCache) {
 	p.dag = dag
 	p.source = source
 
-	// NOT IMPLEMENTED
+	// reset the processor when we're done
+	defer func() {
+		p.dag = nil
+		p.source = nil
+	}()
+
+	for _, route := range p.validHTTPRoutes() {
+		p.computeHTTPRoute(route)
+	}
+}
+
+func (p *ServiceAPIsProcessor) computeHTTPRoute(route *serviceapis.HTTPRoute) {
+
+	// Determine the hosts on the route, if no hosts
+	// are defined, then set to "*".
+	var hosts []string
+	if len(route.Spec.Hostnames) == 0 {
+		hosts = append(hosts, "*")
+	} else {
+		for _, host := range route.Spec.Hostnames {
+			hosts = append(hosts, string(host))
+		}
+	}
+
+	for _, rule := range route.Spec.Rules {
+
+		// Paths are a set of paths that are
+		var pathPrefixes []string
+		var services []*Service
+
+		for _, match := range rule.Matches {
+			switch match.Path.Type {
+			case serviceapis.PathMatchPrefix:
+				pathPrefixes = append(pathPrefixes, stringOrDefault(match.Path.Value, "/"))
+			default:
+				p.Error("NOT IMPLEMENTED: Only PathMatchPrefix is currently implemented.")
+			}
+		}
+
+		for _, forward := range rule.ForwardTo {
+			// Verify the service is valid
+			if forward.ServiceName == nil {
+				p.Error("ServiceName must not specified and is currently only implemented!")
+				break
+			}
+			meta := types.NamespacedName{Name: *forward.ServiceName, Namespace: route.Namespace}
+
+			// TODO: Refactor EnsureService to take an int32 so conversion to intstr is not needed.
+			service, err := p.dag.EnsureService(meta, intstr.FromInt(int(forward.Port)), p.source)
+			if err != nil {
+				// TODO: Raise `ResolvedRefs` condition on Gateway with `DegradedRoutes` reason.
+				return
+			}
+			services = append(services, service)
+		}
+
+		routes := p.route(pathPrefixes, services)
+
+		for _, vhost := range hosts {
+			vhost := p.dag.EnsureVirtualHost(vhost)
+			for _, route := range routes {
+				vhost.addRoute(route)
+			}
+		}
+	}
+}
+
+// route builds a []*dag.Route for the supplied HTTPRoute.
+func (p *ServiceAPIsProcessor) route(pathPrefixes []string, services []*Service) []*Route {
+	var clusters []*Cluster
+	var routes []*Route
+
+	for _, service := range services {
+		clusters = append(clusters, &Cluster{
+			Upstream: service,
+			Protocol: service.Protocol,
+		})
+	}
+
+	for _, prefix := range pathPrefixes {
+		r := &Route{
+			Clusters: clusters,
+		}
+		r.PathMatchCondition = &PrefixMatchCondition{Prefix: prefix}
+		routes = append(routes, r)
+	}
+
+	return routes
+}
+
+// httpRoutes returns a slice of *serviceapis.HTTPRoute objects.
+// invalid HTTPRoute objects are excluded from the slice and their status
+// updated accordingly.
```

Comment:
s/Invalid/invalid/

I don't see where status is being updated. If the status update occurs elsewhere, I suggest removing "and their status updated accordingly."

Reply:
I'll update the comment, currently this PR doesn't have status implemented, just trying to get a baseline set of code in opening up more features.